frontend/src/e2e-playwright/pages/citizen/citizen-calendar.ts

```diff
@@ -0,0 +1,142 @@
+    private readonly type: 'desktop' | 'mobile'
+  ) {}
+
+  #openCalendarActionsModal = this.page.locator(
+    '[data-qa="open-calendar-actions-modal"]'
+  )
+
+  #childCheckbox = (childId: string) =>
+    new Checkbox(this.page, `[data-qa="child-${childId}"]`)
+  #startDateInput = new RawTextInput(this.page, '[data-qa="start-date"]')
+  #endDateInput = new RawTextInput(this.page, '[data-qa="end-date"]')
+  #repetitionCombobox = new Combobox(this.page, '[data-qa="repetition"]')
+  #dailyStartTimeInput = new RawTextInput(
+    this.page,
+    '[data-qa="daily-start-time"]'
+  )
+  #dailyEndTimeInput = new RawTextInput(this.page, '[data-qa="daily-end-time"]')
+  #weeklyStartTimeInputs = [0, 1, 2, 3, 4, 5, 6].map(
+    (index) =>
+      new RawTextInput(this.page, `[data-qa="weekly-start-time-${index}"]`)
+  )
+  #weeklyEndTimeInputs = [0, 1, 2, 3, 4, 5, 6].map(
+    (index) =>
+      new RawTextInput(this.page, `[data-qa="weekly-end-time-${index}"]`)
+  )
+  #absenceChip = (type: string) =>
+    new Checkbox(this.page, `[data-qa="absence-${type}"]`)
+  #modalSendButton = this.page.locator('[data-qa="modal-okBtn"]')
+
+  #dayCell = (date: LocalDate) =>
+    this.page.locator(
+      `[data-qa="${this.type}-calendar-day-${date.formatIso()}"]`
+    )
+
+  async createReperatingDailyReservation(
+    dateRange: FiniteDateRange,
+    startTime: string,
+    endTime: string
+  ) {
+    if (this.type === 'mobile') {
+      await this.#openCalendarActionsModal.click()
+      await this.page
+        .locator('[data-qa="calendar-action-reservations"]')
+        .click()
+    } else {
+      await this.page.locator('[data-qa="open-reservations-modal"]').click()
+    }
+
+    await this.#startDateInput.clear()
+    await this.#startDateInput.type(dateRange.start.format())
+    await this.#endDateInput.type(dateRange.end.format())
+    await this.#dailyStartTimeInput.type(startTime)
+    await this.#dailyEndTimeInput.type(endTime)
+
+    await this.#modalSendButton.click()
+  }
+
+  async createReperatingWeeklyReservation(
+    dateRange: FiniteDateRange,
+    weeklyTimes: { startTime: string; endTime: string }[]
+  ) {
+    if (this.type === 'mobile') {
+      await this.#openCalendarActionsModal.click()
+      await this.page
+        .locator('[data-qa="calendar-action-reservations"]')
+        .click()
+    } else {
+      await this.page.locator('[data-qa="open-reservations-modal"]').click()
+    }
+
+    await this.#startDateInput.clear()
+    await this.#startDateInput.type(dateRange.start.format())
+    await this.#endDateInput.type(dateRange.end.format())
+    await this.#repetitionCombobox.fill('Viikoittain')
+    await this.#repetitionCombobox.findItem('Viikoittain').click()
+    await weeklyTimes.reduce(async (promise, { startTime, endTime }, index) => {
+      await promise
+      await this.#weeklyStartTimeInputs[index].type(startTime)
+      await this.#weeklyEndTimeInputs[index].type(endTime)
+    }, Promise.resolve())
```

Comment:
What's the purpose of reduce over e.g. forEach or a basic for-loop? The function also doesn't return anything, so on first item it will `await Promise.resolve()` and after that it will `await undefined`, which also confuses me.

Reply:
Async functions always return promises. So that function returns `Promise<void>`. So in effect we wait for the previous start and end times to be written into the inputs before taking the next weekday.